### PR TITLE
Add support for the OV5640 to the Arduino Portenta

### PR DESCRIPTION
**Pull request by kwagyeman**
Sets the PCLK to 70 MHz and gets it working.

### CODE REVIEW

File: src/omv/sensors/ov5640.h [hunk 1]
```diff
@@ -10,6 +10,10 @@
  */
 #ifndef __OV5640_H__
 #define __OV5640_H__
+#if defined(PORTENTA)
+#define OV5640_XCLK_FREQ 20000000
+#else
 #define OV5640_XCLK_FREQ 24000000
+#endif
```
Comment (iabdalkader) [inline src/omv/sensors/ov5640.h:17]:
```suggestion
```
Define in omv_boardconfig.h or switch all to 20MHz.

File: src/omv/boards/PORTENTA/omv_boardconfig.h [hunk 2]
```diff
@@ -48,8 +48,8 @@
 #define OMV_HARDWARE_JPEG       (1)
 
 // Enable sensor drivers
-#define OMV_ENABLE_OV2640       (0)
-#define OMV_ENABLE_OV5640       (0)
+#define OMV_ENABLE_OV2640       (1)
```
Comment (iabdalkader) [inline src/omv/boards/PORTENTA/omv_boardconfig.h:51]:
```suggestion
#define OMV_ENABLE_OV2640       (0)
```

File: src/omv/sensors/ov5640.c [hunk 3]
```diff
@@ -115,8 +115,13 @@ static const uint8_t default_regs[][3] = {
     { 0x38, 0x15, 0x31 },
     { 0x30, 0x34, 0x1a },
     { 0x30, 0x35, 0x11 }, // { 0x30, 0x35, 0x21 },
-    { 0x30, 0x36, 0x64 }, // { 0x30, 0x36, 0x46 },
+    #if defined(PORTENTA)
+    { 0x30, 0x37, 0x12 },
+    { 0x30, 0x36, 0x46 },
+    #else
```
Comment (iabdalkader) [inline src/omv/sensors/ov5640.c:121]:
```suggestion
    #if (OV5640_XCLK_FREQ==20000000)
    { 0x30, 0x37, 0x12 },
    { 0x30, 0x36, 0x46 },
    #else
```

File: src/omv/sensors/ov5640.c [hunk 4]
```diff
@@ -665,7 +670,7 @@ static int reset(sensor_t *sensor)
         ret |= cambus_writeb2(&sensor->bus, sensor->slv_addr, (default_regs[i][0] << 8) | (default_regs[i][1] << 0), default_regs[i][2]);
     }
 
-    #if defined(MCU_SERIES_H7)
+    #if defined(MCU_SERIES_H7) && (!defined(PORTENTA))
```
Comment (iabdalkader) [inline src/omv/sensors/ov5640.c:673]:
```suggestion
    #if defined(MCU_SERIES_H7) && (OV5640_XCLK_FREQ == 24000000)
```

**Review by iabdalkader — CHANGES_REQUESTED** [review]

**Comment by kwagyeman** [comment]
Work in progress commit. Restructured code to meet previous comments. Everything is defined in headers now.

That said, OV5640 camera appears to pass probe and the clock is set to the right freq... however, all I2C accesses after the probe fail. Not sure what's wrong. Will need to debug later after reviewing new designs.

**Comment by kwagyeman** [comment]
This is mergeable now. Portena OV5640 works at 30 FPS.

**Comment by kwagyeman** [comment]
Merge please.

**Comment by iabdalkader** [comment]
This pull request has been mentioned on **OpenMV Forums**. There might be relevant details there:

https://forums.openmv.io/t/portenta-better-camera/10565/5

**Pull request merged**